Modules/IO/ImageBase/include/itkConvertPixelBuffer.hxx

```diff
@@ -481,6 +490,33 @@ ConvertPixelBuffer<InputPixelType, OutputPixelType, OutputConvertTraits>::Conver
   }
 }
 
+template <typename InputPixelType, typename OutputPixelType, typename OutputConvertTraits>
+void
+ConvertPixelBuffer<InputPixelType, OutputPixelType, OutputConvertTraits>::ConvertVectorToVector(
+  InputPixelType *  inputData,
+  int               inputNumberOfComponents,
+  OutputPixelType * outputData,
+  size_t            size)
+{
+  int outputNumberOfComponents = OutputConvertTraits::GetNumberOfComponents();
+  int componentCount = std::min(inputNumberOfComponents, outputNumberOfComponents);
+
+  for (size_t i = 0; i < size; ++i)
+  {
+    for (int c = 0; c < componentCount; ++c)
+    {
+      OutputConvertTraits::SetNthComponent(c, *outputData, static_cast<OutputComponentType>(*(inputData + c)));
+    }
+    for (int c = componentCount; c < outputNumberOfComponents; ++c)
+    {
+      OutputConvertTraits::SetNthComponent(c, *outputData, 0); // set the rest of components to zero
+    }
+
+    ++outputData;
+    inputData += inputNumberOfComponents;
```

Comment:
It looks like these two lines are both advancing to the next respective input/output pixel. Just to confirm, is there a reason for the difference in notation?

Reply:
I was following the pattern established by other similar code in that file.